packages/common/src/featureServiceHelpers.ts

```diff
@@ -926,6 +926,9 @@ export function addFeatureServiceDefinition(
         _updateTemplateDictionaryFields(itemTemplate, templateDictionary);
       }
 
+      const isSelfReferential = _isSelfReferential(listToAdd);
+      listToAdd = _updateOrder(listToAdd, isSelfReferential);
```

Comment:
@MikeTschudi I needed to pass the result of _isSelfReferential to two functions so only wanted to call it once.

Reply:
@MikeTschudi just read the question again...my bad. The reason is because this approach does not add any additional if branches for the tests of the more complicated addFeatureServiceDefinition function. So I can just write a small focused test for the new function.